### PR TITLE
Refactor OOI list Filtering

### PR DESCRIPTION
**Pull request by Rieven**
### Changes
- All checkboxes on removed (only check what you want to filter)
- Remove double 'Set Filters' button, added also 'Clear Filter' like Finding List 
- Refactor OOI-type filtering using Django forms to populate checkboxes instead of dictionary addup
- 'Currently filter on' label gives more information what is filtered. 
- Active filters is also available in exported json format
- Active filters is also available in exported CSV format 

### Issue link
**Issue nr 3** in: https://github.com/minvws/nl-kat-coordination/issues/1666

Closes https://github.com/minvws/nl-kat-coordination/issues/1666 **(Issue 3)**

### Proof
image

Active filters text:
image

Filter options
image

Filters shows in exported json:
image

Exported CSV:
image

---

### Code Checklist
- [x] All the commits in this PR are properly PGP-signed and verified;
- [x] This PR only contains functionality relevant to the issue; tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

### Communication
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have made corresponding changes to the documentation, if necessary.

---
## Checklist for code reviewers:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---
## Checklist for QA:
Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

File: rocky/rocky/templates/partials/form/field_input_multiselect.html [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
-            <input type="{% if choice.data.value < 0 %}hidden{% else %}checkbox{% endif %}"
-                   name="{{ field.name }}"
-                   value="{{ choice.data.value }}"
-                   id="id_{{ field.name }}_{{ forloop.counter0 }}"
-                   {% if choice.data.attrs.disabled %}disabled{% endif %}
-                   {% if choice.data.attrs.checked %}checked{% endif %} />
-            <label for="id_{{ field.name }}_{{ forloop.counter0 }}">{{ choice.choice_label|capfirst }}</label>
-        </div>
-    {% endfor %}
+    <div class="{{ custom_class }}">
```
Comment (Rieven) [inline rocky/rocky/templates/partials/form/field_input_multiselect.html:3]: only added this line

**Comment by underdarknl** [comment]
In the expanded filter options menu no options are selected, is that because no options where selected, or should the options that are present in the 'Active filters text:' screenshot be active there too?

**Review by dekkers — COMMENTED** [review]
Looks good. :+1:

**Comment by Rieven** [comment]
> In the expanded filter options menu no options are selected, is that because no options where selected, or should the options that are present in the 'Active filters text:' screenshot be active there too?

Oww those are 2 different screenshots. Indeed what is selected in the filter option is also shown as text.

File: rocky/rocky/templates/organizations/organization_crisis_room.html [hunk 2]
```diff
@@ -26,7 +26,8 @@ <h1>{% translate "Crisis room" %}: {{ organization.name }}</h1>
                     {% translate "An overview of the top 10 most severe findings OpenKAT found. Check the detail section for additional severity information." %}
                 </p>
                 <h2>{% translate "Top 10 most severe Findings" %}</h2>
-                {{ object_type_filters }}
+                {% include "partials/show_active_filters.html" %}
+
```
Comment (dekkers) [inline rocky/rocky/templates/organizations/organization_crisis_room.html:30]: There is no filter on this page. `object_type_filter` didn't do anything because it was not set before, and the partial also won't do anything because `active_filters` isn't defined. Better to just remove this from the template.

File: rocky/rocky/templates/findings/finding_list.html [hunk 3]
```diff
@@ -17,7 +17,7 @@ <h1>{% translate "Findings @ " %}{{ valid_time|date }}</h1>
                         Each finding relates to an object. Click a finding for additional information.
                     {% endblocktranslate %}
                 </p>
-                {{ object_type_filters }}
+                {% include "partials/show_active_filters.html" %}
```
Comment (dekkers) [inline rocky/rocky/templates/findings/finding_list.html:20]: `object_type_filters` wasn't defined and this view also doesn't define `active_filters`. Better to remove this completely. If we want to change this to show active filters we can better do this in another PR. Before writing that code we should probably first think about how we can structure the views show we don't have to duplicate things.

**Comment by Darwinkel** [comment]
## Checklist for QA:
- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
    - [x] I have managed to pass the onboarding flow
    - [x] Objects and Findings are created properly
    - [x] Tasks are created and completed properly
- [ ] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.

### What works:
* UI and JSON/CSV exports are correct when using a single filter
image
image
image

* UI and JSON/CSV exports are correct when using the OOI and the vrijwaring filters
image

* Exporting and filtering still works correctly overall

### What doesn't work:
* It looks like filtering by inheritance as well as clearance results in clearance being overridden
image
image
image

### Bug or feature?:
- When exporting to JSON/CSV, the filters are in the KAT interface language. Is this intended? 
image

**Comment by Rieven** [comment]
> ## Checklist for QA:
> * [x]  I have checked out this branch, and successfully ran a fresh `make reset`.
> * [x]  I confirmed that there are no unintended functional regressions in this branch:
>   
>   * [x]  I have managed to pass the onboarding flow
>   * [x]  Objects and Findings are created properly
>   * [x]  Tasks are created and completed properly
> * [ ]  I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
> 
> ### What works:
> * UI and JSON/CSV exports are correct when using a single filter
>   image
>   image
>   image
> * UI and JSON/CSV exports are correct when using the OOI and the vrijwaring filters
>   image
> * Exporting and filtering still works correctly overall
> 
> ### What doesn't work:
> * It looks like filtering by inheritance as well as clearance results in clearance being overridden
>   image
 Fixed in: https://github.com/minvws/nl-kat-coordination/pull/1796/commits/1dc61c3ea6225e3eba9e546d71d7c148fce52bcb (wrong translation)

**Comment by Darwinkel** [comment]
Fixes confirmed. Ready to merge.

**Review by dekkers — APPROVED** [review]
